DetectorDescription/DDCMS/plugins/dd4hep/DDDefinitions2Objects.cc

```diff
@@ -947,8 +947,12 @@ void Converter<PartSelector>::operator()(xml_h element) const {
            "+++ PartSelector for %s path: %s",
            specParName.c_str(),
            path.c_str());
-  if (path.substr(0, 3) == std::string(".*:")) {
-    path = path.substr(3);
+  auto npos = path.find(':');
+  if (npos != path.npos) {
+    std::string anyns{"//.*"};
+    if (path.substr(0, npos) == anyns) {
+      path = string("//").append(path.substr(npos + 1));
+    }
   }
   registry.specpars[specParName].paths.emplace_back(path);
```

Comment:
```suggestion
  registry.specpars[specParName].paths.emplace_back(std::move(path));
```
might avoid a copy.

Reply:
A quick test in compiler explorer shows a the `std::move()` indeed leads to fewer generated instructions.